www/index.js

```diff
@@ -739,10 +742,29 @@ function init() {
     show("#details_custom");
   }
 
-  setupSelectList($("#versions"), Object.keys(config.versions), (version) => {
+  let versions = await fetch(config.image_url + "/.versions.json", { cache: "no-cache" })
+    .then((obj) => { return obj.json(); })
+    .then((obj) => { return obj.versions_list.filter((version) => version.localeCompare("19.07.4") >= 0);});
+
+  if (config.show_snapshots) {
+    let branches = [];
+    for (const version of versions.slice()) {
+      let branch = version.split('.').slice(0, -1).join('.') + '-SNAPSHOT';
+      if (!versions.includes(branch)) {
+        versions.push(branch);
+      }
+    }
+    versions.push("SNAPSHOTS");
+  }
+
+  setupSelectList($("#versions"), versions, (version) => {
     // A new version was selected
-    let base_url = config.versions[version];
-    fetch(base_url + "/overview.json", { cache: "no-cache" })
+    let base_url = `${config.image_url}/releases/${version}`;
+    if (version == "SNAPSHOTS") {
```

Comment:
This is quiet a hack. Can we avoid to make this project too OpenWrt specific?

Reply:
Sure, what do you suggest?